### PR TITLE
remote: 'grpc' scheme prefix converted to an empty scheme for grpc remote connections

### PR DESCRIPTION
**Pull request by ishikhman**
Part of #7205

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/authandtls/GoogleAuthUtils.java [hunk 1]
```diff
@@ -70,10 +72,22 @@ public static ManagedChannel newChannel(String target, AuthAndTLSOptions options
...
+  /**
+   * Converts 'grpc(s)' into an empty protocol,
+   * because 'grpc(s)' is not a widely supported scheme and is interpreted as 'dns' under the hood.
+   *
+   * @return target URL with converted scheme
+   */
+  private static String convertTargetScheme(String target) {
+    return target
+        .replace("grpc://", "")
+        .replace("grpcs://", "");
```
Comment (buchgr) [inline src/main/java/com/google/devtools/build/lib/authandtls/GoogleAuthUtils.java]: add a TODO to implement grpcs?

File: src/main/java/com/google/devtools/build/lib/authandtls/GoogleAuthUtils.java [hunk 2]
```diff
@@ -53,9 +53,11 @@ public static ManagedChannel newChannel(String target, AuthAndTLSOptions options
     final SslContext sslContext =
         options.tlsEnabled ? createSSlContext(options.tlsCertificate) : null;
 
+    String targetUrl = convertTargetScheme(target);
+
     try {
       NettyChannelBuilder builder =
-          NettyChannelBuilder.forTarget(target)
+          NettyChannelBuilder.forTarget(targetUrl)
               .negotiationType(options.tlsEnabled ? NegotiationType.TLS : NegotiationType.PLAINTEXT)
```
Comment (buchgr) [inline src/main/java/com/google/devtools/build/lib/authandtls/GoogleAuthUtils.java]: implement grpcs in this PR?

Comment (ishikhman) [reply src/main/java/com/google/devtools/build/lib/authandtls/GoogleAuthUtils.java]: I'll add it as a next PR, just to keep it separate.

Comment (buchgr) [reply src/main/java/com/google/devtools/build/lib/authandtls/GoogleAuthUtils.java]: it's a one liner

**Comment by buchgr** [comment]
LGTM

File: src/main/java/com/google/devtools/build/lib/authandtls/GoogleAuthUtils.java (hunk 2 above)
Comment (ishikhman) [reply src/main/java/com/google/devtools/build/lib/authandtls/GoogleAuthUtils.java]: not really :)

Comment (buchgr) [reply src/main/java/com/google/devtools/build/lib/authandtls/GoogleAuthUtils.java]: options.tlsEnabled || target.startsWith("grpcs") ? ...

**Pull request closed without merging**